### PR TITLE
iconUrl -> icon

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes https://github.com/dotnet/roslyn-analyzers/issues/5054

### CODE REVIEW

**Comment by mmitche** [comment]
@Youssef1313 The icon element is to an embedded icon, rather than a url.

File: eng/GenerateAnalyzerNuspec.targets [hunk 1]
```diff
@@ -144,7 +144,7 @@
       <_NuspecMetadata Include="copyright=$(Copyright)" />
       <_NuspecMetadata Include="license=$(PackageLicenseExpression)" />
       <_NuspecMetadata Include="projectUrl=$(PackageProjectUrl)" />
-      <_NuspecMetadata Include="iconUrl=$(PackageIconUrl)" />
+      <_NuspecMetadata Include="icon=$(PackageIconUrl)" />
```
Comment (Youssef1313) [inline eng/GenerateAnalyzerNuspec.targets:147]:
```suggestion
      <_NuspecMetadata Include="icon=$(PackageIcon)" />
```

**Comment by Youssef1313** [comment]
Thanks @mmitche. I updated the property name. Since no packages in the repository uses an icon, I believe no other renames are needed. But will let @mavasani confirm.

**Comment by mmitche** [comment]
I think you'll also need to ensure that you package up the icon too. There is one being brought in from arcade (PackageIconFullPath), but the build and packaging logic in this repo is different enough from arcade that it might take some doing to get it to work.

File: src/Tools/GenerateAnalyzerNuspec/Program.cs [hunk 2]
```diff
@@ -85,6 +85,7 @@
 result.AppendLine(@"  </metadata>");
 
 result.AppendLine(@"  <files>");
+result.AppendLine(@"    $CommonFileElements$");
```
Comment (Youssef1313) [inline src/Tools/GenerateAnalyzerNuspec/Program.cs:88]: Added this based on the description in https://github.com/dotnet/arcade/pull/4078.

**Comment by mmitche** [comment]
@Youssef1313 Did you check the package content to verify it has the expected nuspec and icon file?

**Comment by Youssef1313** [comment]
@mmitche I ran `build.cmd -pack`, changed one of the produced nupkg to `.zip` and extracted it, then verified `Icon.png`:

image

Anything else to check?

**Comment by mmitche** [comment]
LGTM then!

**Review by mmitche — APPROVED** [review]

**Pull request merged**